### ISSUE TITLE
Consolidate synchronous and asynchronous fetching in DB manager

### ISSUE THREAD

**Issue by manuelma** [spine-tools/Spine-Toolbox#1838]
Re https://github.com/Spine-project/Spine-Toolbox/issues/1834

I think we need to do some work to clean up SpineDBManager - SpineDBSignaller - SpineDBWorker code. The first time we implemented asynchronous/lazy/incremental fetching we kept some interfaces that were built for synchronous/single-shot fetching (i.e. fetching the entire DB at the moment of establishing the DB connection). I guess we didn't want to change too much of the base code at once, so we awkwardly adapted the async fetching to work with something that was only intended for synch fetching (SpineDBSignaller, mainly). So now I think we have a lot of redundant code. We should readapt the SpineDBManager interface to work entirely with asynch fetching and I think we would be able to simplify/remove a lot of code - that currently makes debugging kinda hard.

**Comment by manuelma** [comment]
https://github.com/Spine-project/Spine-Toolbox/pull/1853 has now been merged. Most of the changes (99%) are on how things work behind the scenes.

The biggest change is expanding a view doesn't affect other views. Each view is populated as expanded and there are no side effects on other views. Before, expanding a view would collect items from the DB and as those items were fetched, *all* views that could see those elements would be populated too - even though the user expanded just one of them. The reason was that at the beginning, we wanted to load the entire DB editor at once, so we would fecth all items from the DB and populate all the views at once. When we introduced lazy loading, we would fetch items incrementally from the DB but still populate all views as those items were fetched. Now we fetch items incrementally and only populate the view that requested the items.

I think this change might have advantages from the user experience point of view, but the most important thing to me is the code became simpler (at least in my opinion). Since views do not need to react to exogenous fetches, fetching and populating can be performed as a single operation by a single actor (a `FetchParent` instance). Previously, fetching and populating were completely disconnected - a view would submit a fetch more request and forget all about it - which meant a lot of extra work figuring out if the view that requested the items actually need to show them!

There is still more work to make the item metadata editor and a lot of dialogs and delegates lazy - they are still non-lazy for legacy reasons.
